sktime/forecasting/base/adapters/_statsmodels.py

```diff
@@ -111,12 +111,15 @@ def _predict(self, fh, X):
         # statsmodels exog must contain test indices only.
         # For discussion see https://github.com/sktime/sktime/issues/3830
         if X is not None:
-            ind_drop = self._X.index
-            X = X.loc[~X.index.isin(ind_drop)]
+            if self._X is not None:
+                ind_drop = self._X.index
```

Comment:
This is the question I asked yesterday. Is it guaranteed that `ind_drop` only contains indices from training range and never from prediction range?

Reply:
Here `ind_drop` contains the indices present in the `X` passed to `fit` (which should be the training set ideally).

Reply:
Here `ind_drop` contains the indices present in the `X` passed to `fit` (which should be the training set ideally).

Reply:
Ideally and more common, yes. Not sure if it's a guarantee. For example, I personally have definitely passed full `X` sometimes (not necessarily to SARIMAX, but I don't remember exactly).

Reply:
Ok so in the event that a user passes full `X` to SARIMAX in fit , what will they pass to predict? If it is a subset of this `X`, (say `test_X` obtained by splitting `X`) it will result in all it's indices being dropped and the code will run without error but the exog features will not be used in this case. However ideally the user should be passing different X to fit and predict in order for it to serve it's purpose.
This code to drop indices was added as a fix for issue #3830 as mentioned in the comment.
https://github.com/sktime/sktime/blob/14c88c69d2a0e961390e749278b5349470d35332/sktime/forecasting/base/adapters/_statsmodels.py#L110-L115
We technically cannot do anything about `full X` being passed to `fit` because we do not know what the user will pass to `predict` next.

Reply:
> it will result in all it's indices being dropped and the code will run without error

Can you please validate this? My intinct says that the underlying model (from `statsmodels`) will complain if `X` is not passed during `predict` but passed during `fit`.

Reply:
>it will result in all it's indices being dropped and the code will run without error

Small correction here, I meant this in the case where `full X` is passed to `fit` and a `subset of X` is passed to `predict`. However I was wrong in saying that code runs without error, rather all indices get dropped and it returns an `error` saying the received exog is not in the correct shape.
```python
from sktime.datasets import load_longley
from sktime.split import temporal_train_test_split
from sktime.forecasting.sarimax import SARIMAX

y, X = load_longley()
y_train, _, X_train, X_test = temporal_train_test_split(y, X)
forecaster = SARIMAX()
forecaster.fit(y,X=X)
forecaster.predict(fh=[1, 2, 3, 4],X=X_train)
```
```python
ValueError: Provided exogenous values are not of the appropriate shape. Required (4, 5), got (0, 5).
```
Also, if X is passed to `fit` but not `predict`, it returns the following error asking for exog features to be passed in predict as well.
```python
ValueError: Out-of-sample operations in a model with a regression component require additional exogenous values via the `exog` argument.
```

Reply:
@yarnabrina so do we mark this conversation as resolved or are we waiting for @fkiraly /@ciaran-g to review it as well

Reply:
I'd wait for at least 2 days. Otherwise, we can go ahead and merge.

Reply:
Not 100% sure on the tag behaviour myself unfortunately, but the changes look ok and sensible to me 👍